src/app/calculadoras/aguinaldo/page.tsx

```diff
@@ -0,0 +1,200 @@
+'use client'
+import es from 'date-fns/locale/es'
+import {ChangeEvent, useState} from 'react'
+import DatePicker from 'react-datepicker'
+import {Button, Divider, Input, Paragraph} from 'theme-ui'
+
+import {Page} from '~/components/page'
+
+const calculateAguinaldo = (grossSalary: number, daysWorked: number, aguinaldoDays: number): number => {
+  const proportionalAguinaldoDays = (aguinaldoDays / 365) * daysWorked
+  const baseDailySalary = grossSalary / 30.4 // Average days in a month
+
+  return baseDailySalary * proportionalAguinaldoDays
+}
+
+const daysPassedSinceDate = (targetDate: Date): number => {
+  const currentDate = new Date()
+  const timeDifference = currentDate.getTime() - targetDate.getTime()
+  return Math.floor(timeDifference / (1000 * 60 * 60 * 24))
+}
+
+const cleanMoneyFormat = (moneyString: string): number => {
+  // Remove any non-numeric characters except for the decimal point
+  const cleanedString = moneyString.replace(/[^0-9.]/g, '')
+
+  // Parse the cleaned string as a float
+  const numberValue = parseFloat(cleanedString)
+
+  // Check if it's a valid number
+  if (!isNaN(numberValue)) {
+    return numberValue
+  } else {
+    // Handle the case where the input couldn't be parsed as a number
+    return NaN
+  }
+}
+
+/**
+ * This function was obtained from https://codepen.io/559wade/pen/LRzEjj
+ *
+ * Adapted to typescript with ChatGPT
+ *
+ */
+const formatNumber = (n: string): string => {
+  // format number 1000000 to 1,234,567
+  return n.replace(/\D/g, '').replace(/\B(?=(\d{3})+(?!\d))/g, ',')
+}
+
+/**
+ * This function was obtained from https://codepen.io/559wade/pen/LRzEjj
```

Comment:
remove comments, i dont think we will need them

Reply:
Removed